nihms-filesystem-builder/src/main/java/org/dataconservancy/nihms/builder/fs/FilesystemModelBuilder.java

```diff
@@ -18,17 +18,164 @@
+                        journal.setIssn(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_JOURNAL_TITLE:
+                        journal.setJournalTitle(value);
+                        break;
+
+                    //manuscript metadata
+                    case NihmsBuilderPropertyNames.NIHMS_MANUSCRIPT_DOI:
+                        try {
+                            manuscript.setDoi(new URI(value));
+                        } catch (URISyntaxException e1) {
+                            e1.printStackTrace();
+                        }
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_MANUSCRIPT_ID:
+                        manuscript.setNihmsId(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_MANUSCRIPT_PUBMEDID:
+                        manuscript.setPubmedId(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_MANUSCRIPT_PUBMEDCENTRALID:
+                        manuscript.setPubmedCentralId(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_MANUSCRIPT_URL:
+                        manuscript.setManuscriptUrl(new URL(value));
+                        break;
+
+                    //person metadata
+                    case NihmsBuilderPropertyNames.NIHMS_PERSON_AUTHOR:
+                        person.setAuthor(Boolean.parseBoolean(value));
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_PERSON_CORRESPONDINGPI:
+                        person.setCorrespondingPi(Boolean.parseBoolean(value));
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_PERSON_EMAIL:
+                        person.setEmail(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_PERSON_FIRSTNAME:
+                        person.setFirstName(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_PERSON_MIDDLENAME:
+                        person.setMiddleName(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_PERSON_LASTNAME:
+                        person.setLastName(value);
+                        break;
+                    case NihmsBuilderPropertyNames.NIHMS_PERSON_PI:
+                        person.setPi(Boolean.parseBoolean(value));
+                        break;
+                }
+            }
+
+            //now populate the submission
+            files.add(file);
+            submission.setFiles(files);
+
+            persons.add(person);
+            metadata.setPersons(persons);
+
+            metadata.setJournalMetadata(journal);
+            metadata.setManuscriptMetadata(manuscript);
+
+            submission.setMetadata(metadata);
+
+        } catch (IOException ioe) {
+            ioe.printStackTrace();
```

Comment:
You could probably catch `IOException` and wrap it with `InvalidModel`, and then throw the `InvalidModel` exception.  Even though an IOE doesn't mean you have something wrong with the model, it's probably ok to throw `InvalidModel` for now.  

If the semantic mismatch of IOE and InvalidModel bother you enough, you could create a more generic `SubmissionBuilderException` and have `InvalidModel` extend `SubmissionBuilderException`.  Then at this point catch IOE and re-throw it wrapped as the more generic `SubmissionBuilderException`.